### ISSUE TITLE
Deleting an object from S3 bucket does not work

### ISSUE THREAD

**Issue by cowjen01** [keystonejs/keystone#2315]
# Bug report

## Describe the bug

If you try to delete an uploaded file it will only remove a record from the database, but the file remains in S3 bucket.

## To Reproduce

1. Create a model with S3 file adapter
2. Upload file to S3 bucket
3. Now try to delete it
4. Checkout the bucket
5.  Database record id gone but the file remains

## Solution

I found out in `aws-sdk` [docs](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/S3.html#deleteObject-property) that:
> If a callback is not supplied, you must call AWS.Request.send() on the returned request object to initiate the request.

So the `deleteObject` method in [S3 file adapter](https://github.com/keystonejs/keystone/blob/master/packages/file-adapters/lib/s3.js#L79) is missing callback function to really call a request that will remove the file.

I suggest to do something like this:
```js
delete(file, options = {}) {
    return new Promise((resolve, reject) => {
      if (!file) {
        return reject(new Error('Missing required argument "file".'));
      }
      this.s3.deleteObject({
        Bucket: this.bucket,
        Key: path.join(this.folder, file.filename),
        ...options,
      }, (error, data) => {
        if (error) {
          reject(error);
        } else {
          resolve(data);
        }
      });
    });
  }
```

## System information

- OS: macOS
- Browser: chrome

**Comment by LiamAttClarke** [comment]
See: https://github.com/keystonejs/keystone/pull/2291

**Comment by MadeByMike** [comment]
Resolved in #2291
